### PR TITLE
Rust runtime

### PR DESCRIPTION
**Pull request by nebkor**
Make it easy to use regular, safe, Rust functions for entry points.

### CODE REVIEW

File: rust/src/bin/timer.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#![no_std]
+#![no_main]
+
+// Copyright 2022, Brian Swetland <swetland@frotz.net>
+// Licensed under the Apache License, Version 2.0
+
+/*
```
Comment (swetland) [inline rust/src/bin/timer.rs:15]: Did you mean to just check in a copy of the original commented out?

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@ out/
 .*.swp
 .*.swo
 local.mk
+.dir-locals.el
```
Comment (swetland) [inline .gitignore:5]: no problem with this, but I prefer checkins be logically grouped (eg try to not combine meta/config stuff with code, or code with build, or whatnot unless it cannot be cleanly done as separate changes)

**Review by swetland — CHANGES_REQUESTED** [review]

File: rust/src/bin/timer.rs (hunk 1 above)
Comment (nebkor) [reply rust/src/bin/timer.rs:15]: Kind of :) I started to port it but realized I wanted to do some house-cleaning first.

Comment (nebkor) [reply rust/src/bin/timer.rs]: Removed.

File: .gitignore (hunk 2 above)
Comment (nebkor) [reply .gitignore:4]: Fixed.

**Pull request merged**